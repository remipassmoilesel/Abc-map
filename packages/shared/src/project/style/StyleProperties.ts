/**
 * Copyright © 2023 Rémi Pace.
 * This file is part of Abc-Map.
 *
 * Abc-Map is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of
 * the License, or (at your option) any later version.
 *
 * Abc-Map is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General
 * Public License along with Abc-Map. If not, see <https://www.gnu.org/licenses/>.
 */

export enum StyleProperties {
  FillPattern = 'abc:style:fill:pattern',
  FillColor1 = 'abc:style:fill:color1',
  FillColor2 = 'abc:style:fill:color2',
  StrokeColor = 'abc:style:stroke:color',
  StrokeWidth = 'abc:style:stroke:width',
  TextValue = 'abc:style:text:value',
  TextColor = 'abc:style:text:color',
  TextSize = 'abc:style:text:size',
  TextFont = 'abc:style:text:font',
  TextOffsetX = 'abc:style:text:offset-x',
  TextOffsetY = 'abc:style:text:offset-y',
  TextRotation = 'abc:style:text:rotation',
  TextAlignment = 'abc:style:text:alignment',
  PointIcon = 'abc:style:point:icon',
  PointSize = 'abc:style:point:size',
  PointColor = 'abc:style:point:color',
  ZIndex = 'abc:style:z-index',
}

export enum FillPatterns {
  Flat = 'abc:style:fill:flat',
  Circles = 'abc:style:fill:circles',
  Squares = 'abc:style:fill:squares',
  HatchingVertical = 'abc:style:fill:hatching:vertical',
  HatchingHorizontal = 'abc:style:fill:hatching:horizontal',
  HatchingObliqueRight = 'abc:style:fill:hatching:oblique:right',
  HatchingObliqueLeft = 'abc:style:fill:hatching:oblique:left',
}
