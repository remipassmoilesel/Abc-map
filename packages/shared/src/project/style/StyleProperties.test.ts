/**
 * Copyright © 2023 Rémi Pace.
 * This file is part of Abc-Map.
 *
 * Abc-Map is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of
 * the License, or (at your option) any later version.
 *
 * Abc-Map is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General
 * Public License along with Abc-Map. If not, see <https://www.gnu.org/licenses/>.
 */

import { FillPatterns, StyleProperties } from './StyleProperties';

/**
 * If this test fail, you should write a migration script then adapt it
 */
describe('StyleProperties', () => {
  it('StyleProperties should not change without migration', () => {
    expect(StyleProperties).toMatchSnapshot();
  });

  it('FillPatterns should not change without migration', () => {
    expect(FillPatterns).toMatchSnapshot();
  });
});
