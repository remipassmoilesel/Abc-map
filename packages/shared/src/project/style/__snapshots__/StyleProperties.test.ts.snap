// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyleProperties FillPatterns should not change without migration 1`] = `
Object {
  "Circles": "abc:style:fill:circles",
  "Flat": "abc:style:fill:flat",
  "HatchingHorizontal": "abc:style:fill:hatching:horizontal",
  "HatchingObliqueLeft": "abc:style:fill:hatching:oblique:left",
  "HatchingObliqueRight": "abc:style:fill:hatching:oblique:right",
  "HatchingVertical": "abc:style:fill:hatching:vertical",
  "Squares": "abc:style:fill:squares",
}
`;

exports[`StyleProperties StyleProperties should not change without migration 1`] = `
Object {
  "FillColor1": "abc:style:fill:color1",
  "FillColor2": "abc:style:fill:color2",
  "FillPattern": "abc:style:fill:pattern",
  "PointColor": "abc:style:point:color",
  "PointIcon": "abc:style:point:icon",
  "PointSize": "abc:style:point:size",
  "StrokeColor": "abc:style:stroke:color",
  "StrokeWidth": "abc:style:stroke:width",
  "TextAlignment": "abc:style:text:alignment",
  "TextColor": "abc:style:text:color",
  "TextFont": "abc:style:text:font",
  "TextOffsetX": "abc:style:text:offset-x",
  "TextOffsetY": "abc:style:text:offset-y",
  "TextRotation": "abc:style:text:rotation",
  "TextSize": "abc:style:text:size",
  "TextValue": "abc:style:text:value",
  "ZIndex": "abc:style:z-index",
}
`;
